### DISCUSSION TITLE
Any mechamism to controll the execution order of middlware?

### DISCUSSION THREAD

**Discussion by GeekEast** [MichalLytek/type-graphql#996] [Q&A]
**Is your feature request related to a problem? Please describe.**
When I have several middlewares registered, I need to control the execution order of them.

**Describe the solution you'd like**
A similar solution in [routing-controller](https://github.com/typestack/routing-controllers#using-middlewares) is that I can add an option called `priority`.

```javascript
// priority: the higher, the earlier
@Middleware({ type: "after", priority: 2 }) 
class ErrorLogger extends ExpressErrorMiddlware {}
```

**Comment by MichalLytek** [comment]
It's not written explicitly in the docs but the order of middlewares in options is the order of calling the middlewares:
https://typegraphql.com/docs/middlewares.html#global-middleware

Same goes to `@UseMiddleware()` usage 😉
